### DISCUSSION TITLE
Typescript Type for the `window. __TAURI__` object.

### DISCUSSION THREAD

**Discussion by FreakeyPlays** [tauri-apps/tauri#11237] [Q&A]
I'm new to Tauri and I'm working on developing both a native application and a web app using a single codebase. However, when using the import statement for Tauri APIs, I encountered errors. To work around this, I decided to use the useGlobalTauri option, accessing the Tauri API through the window object.

Since I'm using TypeScript, I encountered an error that __TAURI__ is not defined on the window object. I resolved this by adding the following type declaration:

```ts
import type * as TauriAPI from '@tauri-apps/api';

declare global {
  interface Window {
    __TAURI__: typeof TauriAPI;
  }
}
```

However, when I started using plugins like dialog or os, I got another error saying those were not defined. Although I know where to find individual types for each function in the Tauri API, is there an automatic way to import all these types together?

(I know there is a [similar discussion](https://github.com/tauri-apps/tauri/discussions/3853) already but it has not been answered since some time)

**Comment by leeonon** [comment]
```ts
import * as app from '@tauri-apps/api';

declare global {
  interface Window {
    __TAURI__: typeof app;
  }
}

```
![CleanShot 2024-11-27 at 18 33 09@2x](https://github.com/user-attachments/assets/f7c422c8-7832-4560-a190-0520579c73f8)

Here is the translation:

It's okay for me to write this way, please check if your node_modules contains the @tauri-apps/api package. 
Version is v2.1.1
